packages/contracts-periphery/contracts/foundry-tests/citizen-house/Badge.t.sol

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.15;
+
+import { Badge } from "../../universal/citizen-house/Badge.sol";
+import { Test } from "forge-std/Test.sol";
+
+contract TBadge is Badge {
+    constructor() Badge("example", "ex", "example.com") {}
+}
+
+contract BadgeTest is Test {
+    TBadge internal badge;
+
+    address testAdr1;
+
+    function setUp() public {
+        badge = new TBadge();
+        testAdr1 = 0x0000008735754EDa8dB6B50aEb93463045fc5c55;
+    }
+
+    function testInvalidMint() public {
+        vm.prank(testAdr1);
+        vm.expectRevert("Error: Sender is not Admin");
+        badge.mint(testAdr1);
+    }
+
```

Comment:
This test file is missing coverage for the majority of the functions on the badge